tests/admin/test_controller.py

```diff
@@ -63,6 +63,7 @@ def test_details(self):
 
         lp.suppressed = False
         with self.app.test_request_context("/"):
+            flask.request.library = self._default_library
```

Comment:
We can get rid of all this code by subclassing test_request_context to set flask.request.library automatically.